mangdangroboticsclub/minipupper_ros
# Update for mini_pupper_control & driver & bringup

Proposed change(s)
------------------

There are three main updates for the Mini Pupper basic keyboard control without Champ:

1.  mini\_pupper\_control This package contains the vel\_to\_servo\_controller which receives the velocity topic /cmd\_vel in ROS2 (and /pose will be added soon) and publishes the joints topic /joint\_command.
    
2.  mini\_pupper\_driver This package contains the servo\_interface and display\_interface. All hardware interface files should be configured in this package.
    
3.  mini\_pupper\_bringup Update the bringup launch file to accommodate the changes in mini\_pupper\_control and mini\_pupper\_driver.
    

### Types of change(s)


Testing and Verification
------------------------

To verify these changes, I performed the following tests:

1.  Run the bringup launch file on Mini Pupper.
2.  Check the state log in the console.
3.  Run the Keyboard node.

Test Configuration
------------------

**Mini Pupper Version**  
Mini Pupper

**Raspberry Pi OS + ROS version**  
Ubuntu 22.04, ROS 2 Humble

**PC OS + ROS version**  
Ubuntu 22.04, ROS 2 Humble

Checklist
---------

*   [x]  I have checked to ensure there aren't other open [Pull Requests](../pulls) for the exact change.

Other comments
--------------

1.  The new mini\_pupper\_bringup package depends on the mini\_pupper\_driver and mini\_pupper\_control packages.
2.  The submodule solution is not suitable because the import part in the Stanford pupper package is complicated. Therefore, the Stanford-related components are placed in mini\_pupper\_control/StanfordQuadruped.
